custom_components/vicare/binary_sensor.py

```diff
@@ -0,0 +1,187 @@
+"""Viessmann ViCare sensor device."""
+from __future__ import annotations
+
+from contextlib import suppress
+from dataclasses import dataclass
+import logging
+
+from PyViCare.PyViCareUtils import (
+    PyViCareInvalidDataError,
+    PyViCareNotSupportedFeatureError,
+    PyViCareRateLimitError,
+)
+import requests
+
+from homeassistant.components.binary_sensor import (
+    DEVICE_CLASS_POWER,
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+
+from . import ViCareRequiredKeysMixin
+from .const import DOMAIN, VICARE_API, VICARE_DEVICE_CONFIG, VICARE_NAME
+
+_LOGGER = logging.getLogger(__name__)
+
+SENSOR_CIRCULATION_PUMP_ACTIVE = "circulationpump_active"
+SENSOR_BURNER_ACTIVE = "burner_active"
+SENSOR_COMPRESSOR_ACTIVE = "compressor_active"
+
+
+@dataclass
+class ViCareBinarySensorEntityDescription(
+    BinarySensorEntityDescription, ViCareRequiredKeysMixin
+):
+    """Describes ViCare binary sensor entity."""
+
+
+GLOBAL_SENSORS: tuple[ViCareBinarySensorEntityDescription, ...] = (
```

Comment:
Also `getCirculationPumpActive` is a circuit sensor. Currently there are no global sensors implemented here, we can add them later ;)

```suggestion
CIRCUIT_SENSORS: tuple[ViCareBinarySensorEntityDescription, ...] = (
```